components/brave_wallet/browser/fil_tx_manager.cc

```diff
@@ -29,27 +41,151 @@ FilTxManager::FilTxManager(TxService* tx_service,
 
 FilTxManager::~FilTxManager() = default;
 
+void FilTxManager::ContinueAddUnapprovedTransaction(
+    const std::string& from,
+    const std::string& tx_meta_id,
+    AddUnapprovedTransactionCallback callback,
+    const std::string& gas_premium,
+    const std::string& gas_fee_cap,
+    int64_t gas_limit,
+    mojom::FilecoinProviderError error,
+    const std::string& error_message) {
+  if (error != mojom::FilecoinProviderError::kSuccess) {
+    std::move(callback).Run(false, "", error_message);
+    return;
+  }
+  std::unique_ptr<FilTxMeta> meta =
+      GetFilTxStateManager()->GetFilTx(tx_meta_id);
+  if (!meta) {
+    DCHECK(false) << "Transaction should be found";
+    std::move(callback).Run(
+        false, "",
+        l10n_util::GetStringUTF8(IDS_BRAVE_WALLET_TRANSACTION_NOT_FOUND));
+    return;
+  }
+
+  meta->tx()->set_gas_premium(gas_premium);
+  meta->tx()->set_fee_cap(gas_fee_cap);
+  meta->tx()->set_gas_limit(gas_limit);
+
+  tx_state_manager_->AddOrUpdateTx(*meta);
+  std::move(callback).Run(true, meta->id(), "");
+}
+
 void FilTxManager::AddUnapprovedTransaction(
     mojom::TxDataUnionPtr tx_data_union,
     const std::string& from,
     const absl::optional<url::Origin>& origin,
-    AddUnapprovedTransactionCallback) {
-  NOTIMPLEMENTED();
+    AddUnapprovedTransactionCallback callback) {
```

Comment:
As mentioned in slack, I don't feel comfortable of this refactoring, please just pass origins for now.

Reply:
done